bib/pkg/progress/progress.go

```diff
@@ -372,14 +372,26 @@ func runOSBuildWithProgress(pb ProgressBar, manifest []byte, store, outputDirect
 		return fmt.Errorf("error starting osbuild: %v", err)
 	}
 	wp.Close()
+	defer func() {
+		// ensure osbuild is stopped even if we exit early
+		if cmd.Process != nil {
+			cmd.Process.Kill()
```

Comment:
This may need a bit more care, maybe only do it in hte error case (as the happy case does a cmd.Wait already so no killing needed)

Reply:
This seems fine as is but I would strongly recommend using PR_SET_PDEATHSIG if it's not already in use. That's a kernel-enforced way to lifecycle bind the child to the parent, such that if e.g. the parent dies due to SIGSEGV or whatever, the child gets killed too.